vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -318,25 +387,34 @@ func (r RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver)
 		for _, mount := range containerJSONInfo.Mounts {
 			// check if the mount location belongs to vsphere plugin
 			if matchNameforVMDK(mount.Source) {
+				// populate temp map
 				r.Incr(mount.Name)
-				log.Debugf("  name=%v (driver=%s source=%s) (%v)",
+				log.Infof("  name=%v (driver=%s source=%s) (%v)",
 					mount.Name, mount.Driver, mount.Source, mount)
 			}
 		}
 	}
 
+	// check if the background refcount was dirtied
+	r.StateMtx.Lock()
+	defer r.StateMtx.Unlock()
+	if r.isDirty == true {
```

Comment:
Please use d.CheckDirty() as in all other places. Also allows to drop L399 and 400